### PR TITLE
Remove safe filter from report templates

### PR DESCRIPTION
**Pull request by mjriley**
## Technical Summary
One of many PRs [removing safe from templates](https://dimagi-dev.atlassian.net/browse/SAAS-12098)

### Safety story
This will be QA'd on staging

### Automated test coverage

None

### QA Plan
Will be releasing several PRs to QA for easier testing.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by biyeun — APPROVED** [review]
great! and thanks for the tests! 🥳

File: corehq/apps/reports/datatables/__init__.py [hunk 1]
```diff
@@ -213,11 +211,13 @@ def render_html(self):
             elif isinstance(column, DataTablesColumnGroup):
                 groups.append(column.render_group_html)
             head.append(column.render_html)
-        head.append("</tr>")
-        groups.append("</tr>")
+
+        headerSegment = format_html_join('\n', '{}', ((segment,) for segment in head))
```
Comment (kaapstorm) [inline corehq/apps/reports/datatables/__init__.py:215]: OCD: 
```suggestion
        header_segment = format_html_join('\n', '{}', ((segment,) for segment in head))
```
(Don't let `aoColumns` drag you down. :upside_down_face: )

File: corehq/apps/reports/datatables/__init__.py [hunk 2]
```diff
@@ -213,11 +211,13 @@ def render_html(self):
...
+        header = format_html('<tr>{}</tr>', headerSegment)
         if len(groups) > 2:
-            head.extend(groups)
-        return "\n".join(head)
+            groupSegment = format_html_join('\n', '{}', ((segment,) for segment in groups))
```
Comment (kaapstorm) [inline corehq/apps/reports/datatables/__init__.py:218]: As above
```suggestion
            group_segment = format_html_join('\n', '{}', ((segment,) for segment in groups))
```

**Review by kaapstorm — APPROVED** [review]
Looks good. Just two obsessive suggestions.

**Review by millerdev — APPROVED** [review]

**Review by biyeun — APPROVED** [review]

**Pull request merged**